pkg/controllers/workgenerator/controller.go

```diff
@@ -220,10 +221,16 @@ func (r *Reconciler) Reconcile(ctx context.Context, req controllerruntime.Reques
 			Message:            "All of the works are synchronized to the latest",
 		})
 		switch {
+		case workDeleted:
+			// Some Work object(s) are being deleted; set a False Applied condition which signals
+			// that resources are in the process of being deleted.
+			klog.V(2).InfoS("Some work being deleted", "resourceBinding", bindingRef)
+			setBindingStatus(workDeleted, works, &resourceBinding)
+			syncErr = controller.NewUserError(fmt.Errorf("some work objects are being deleted"))
 		case !workUpdated:
 			// The Work object itself is unchanged; refresh the cluster resource binding status
 			// based on the status information reported on the Work object(s).
-			setBindingStatus(works, &resourceBinding)
+			setBindingStatus(workDeleted, works, &resourceBinding)
 		case resourceBinding.Spec.ApplyStrategy == nil || resourceBinding.Spec.ApplyStrategy.Type != fleetv1beta1.ApplyStrategyTypeReportDiff:
```

Comment:
update this case and the binding message, to indicate it's deleting works if any and give a concrete example.